### PR TITLE
Task/adq 177 medication jpa model

### PR DESCRIPTION
**Pull request by eclendenning-va**
Added medication entity model and controller support for datamart medication.

Related Stories:
https://vasdvp.atlassian.net/browse/ADQ-177
https://vasdvp.atlassian.net/browse/ADQ-178
https://vasdvp.atlassian.net/browse/ADQ-180

### CODE REVIEW

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medication/MedicationController.java [hunk 1]
```diff
@@ -101,7 +149,8 @@ private CdwMedication101Root search(MultiValueMap<String, String> params) {
 
   /** Search by Identifier. */
   @GetMapping(params = {"identifier"})
-  public Medication.Bundle searchByIdentifier(
+  public Bundle searchByIdentifier(
+      @RequestHeader(value = "Datamart", defaultValue = "") String datamartHeader,
       @RequestParam("identifier") String id,
       @RequestParam(value = "page", defaultValue = "1") @Min(1) int page,
       @CountParameter @Min(0) int count) {
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medication/MedicationController.java]: should delegate to searchById

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medication/MedicationEntity.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class MedicationEntity {
+
+  @Id
+  @Column(name = "CDWId")
+  @EqualsAndHashCode.Include
+  private String cdwId;
+
+  @Column(name = "Medication")
+  @Basic(fetch = FetchType.LAZY)
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medication/MedicationEntity.java]: EAGER

Comment (eclendenning-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medication/MedicationEntity.java]: Coming in the integration test PR 👍

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medication/MedicationController.java (hunk 1 above)
Comment (eclendenning-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medication/MedicationController.java]: Coming to an integration test pr near you ...

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medication/MedicationController.java [hunk 3]
```diff
@@ -116,10 +165,70 @@ private CdwMedication101Root search(MultiValueMap<String, String> params) {
...
+  /**
+   * This class is being used to help organize the code such that all the datamart logic is
+   * contained together. In the future when Mr. Anderson support is dropped, this class can be
+   * eliminated.
+   */
+  private class Datamart {
+
+    private Bundle bundle(
+        MultiValueMap<String, String> parameters, List<Medication> reports, int totalRecords) {
+      PageLinks.LinkConfig linkConfig =
```
Comment (ilaflamme-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medication/MedicationController.java]: Other resources have logging here, like...
`log.info("Search {} found {} results", parameters, entities.getTotalElements());`
Add this for consistency, or you can drop your logging annotation.

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medication/DatamartMedicationControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        ResourceIdentity.builder().system("CDW").resource("MEDICATION").identifier(cdwId).build();
+    when(ids.lookup(publicId)).thenReturn(List.of(resourceIdentity));
+    when(ids.register(Mockito.any()))
+        .thenReturn(
+            List.of(
+                Registration.builder().uuid(publicId).resourceIdentity(resourceIdentity).build()));
+  }
+
+  @Test
+  public void read() {
```
Comment (ilaflamme-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medication/DatamartMedicationControllerTest.java]: I just noticed we have not found tests for readRaw, but not for read.

**Pull request merged**